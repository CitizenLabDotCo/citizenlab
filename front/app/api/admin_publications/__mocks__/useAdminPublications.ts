import { IAdminPublicationData } from '../types';

export const adminPublicationsData: IAdminPublicationData[] = [
  {
    id: 'b3b91b93-e207-4cb0-b49b-737d4b25792e',
    type: 'admin_publication',
    attributes: {
      ordering: 0,
      publication_status: 'published',
      depth: 0,
      publication_title_multiloc: {
        en: 'Quas excepturi ea et.',
        'fr-BE': 'Et illo et aperiam.',
        'nl-BE': 'Praesentium ut itaque ipsam.',
      },
      publication_description_multiloc: {
        en: '\u003cp\u003eAdipisci quia ducimus. Rem et ipsa. Ipsa vero quae.\u003c/p\u003e\u003cp\u003eVoluptas nemo sit. Tempora quia rerum. Ratione dolore quae.\u003c/p\u003e\u003cp\u003eNecessitatibus dolores quod. Magnam ullam natus. Culpa delectus rerum.\u003c/p\u003e',
        'fr-BE':
          '\u003cp\u003eLaudantium et et. Voluptates expedita ipsam. Omnis aut veritatis.\u003c/p\u003e\u003cp\u003eNon cupiditate quia. Illo laudantium tenetur. Neque aperiam aliquid.\u003c/p\u003e\u003cp\u003eAsperiores velit saepe. Veritatis quibusdam est. Fuga laudantium consequatur.\u003c/p\u003e',
        'nl-BE':
          '\u003cp\u003eDeserunt dolor quia. Quae ipsum omnis. Odit autem sint.\u003c/p\u003e\u003cp\u003eEt eos amet. Officia temporibus molestiae. Delectus nihil laboriosam.\u003c/p\u003e\u003cp\u003eMaiores eaque magni. Amet molestias asperiores. Incidunt magni consequatur.\u003c/p\u003e',
      },
      publication_description_preview_multiloc: {
        en: 'Et omnis recusandae minima.',
        'fr-BE': 'Blanditiis et repellendus cupiditate.',
        'nl-BE': 'Libero est voluptatibus tenetur.',
      },
      publication_slug: 'quas-excepturi-ea-et',
      visible_children_count: 1,
    },
    relationships: {
      publication: {
        data: { id: '99d51572-250c-48f4-8950-ea454c9476c1', type: 'folder' },
      },
      parent: { data: null },
      children: {
        data: [
          {
            id: '6012d402-3a56-47a8-8ce6-7cd1c69fa29d',
            type: 'admin_publication',
          },
        ],
      },
    },
  },
  {
    id: '8c8da82d-e4ea-4af4-8c7b-d221e86668cc',
    type: 'admin_publication',
    attributes: {
      ordering: 1,
      publication_status: 'archived',
      depth: 0,
      publication_title_multiloc: {
        en: 'Rem nulla consequatur vero.',
        'fr-BE': 'Ducimus voluptatem tempora expedita.',
        'nl-BE': 'Perferendis modi voluptatem facilis.',
      },
      publication_description_multiloc: {
        en: '\u003cp\u003eAd quam rerum. Fugiat vel dolores. Eligendi occaecati corporis.\u003c/p\u003e\u003cp\u003eDolorem est rem. Sequi aspernatur repudiandae. Tempora reprehenderit aut.\u003c/p\u003e\u003cp\u003eDolor accusantium molestiae. Pariatur aut tempora. Illum sequi illo.\u003c/p\u003e',
        'fr-BE':
          '\u003cp\u003eQuas rem sed. Temporibus corporis et. Fugiat et vero.\u003c/p\u003e\u003cp\u003eEsse aut dolorem. Quis impedit optio. Necessitatibus ipsa ut.\u003c/p\u003e\u003cp\u003eCorporis iure suscipit. Aut autem dolores. Voluptas ea eligendi.\u003c/p\u003e',
        'nl-BE':
          '\u003cp\u003eAliquam amet ut. Ea ut in. Architecto neque minus.\u003c/p\u003e\u003cp\u003eOmnis maxime ullam. Eaque eveniet illum. Animi aut accusantium.\u003c/p\u003e\u003cp\u003eQuasi consectetur dolorem. Ut quam assumenda. Aspernatur pariatur explicabo.\u003c/p\u003e',
      },
      publication_description_preview_multiloc: {
        en: 'Aspernatur ut eius excepturi.',
        'fr-BE': 'Esse incidunt dolore quos.',
        'nl-BE': 'Asperiores possimus labore sequi.',
      },
      publication_slug: 'rem-nulla-consequatur-vero',
      visible_children_count: 1,
    },
    relationships: {
      publication: {
        data: { id: 'c7f4f824-53e1-4b8c-866d-b3412f22d956', type: 'folder' },
      },
      parent: { data: null },
      children: {
        data: [
          {
            id: '8e3a6d01-994e-40d9-9657-9771a100adaf',
            type: 'admin_publication',
          },
        ],
      },
    },
  },
];

export const links = {
  self: 'http://localhost:3000/web_api/v1/admin_publications?depth=0\u0026page%5Bnumber%5D=1\u0026page%5Bsize%5D=1000\u0026publication_statuses%5B%5D=published\u0026publication_statuses%5B%5D=archived\u0026remove_not_allowed_parents=true',
  first:
    'http://localhost:3000/web_api/v1/admin_publications?depth=0\u0026page%5Bnumber%5D=1\u0026page%5Bsize%5D=1000\u0026publication_statuses%5B%5D=published\u0026publication_statuses%5B%5D=archived\u0026remove_not_allowed_parents=true',
  last: 'http://localhost:3000/web_api/v1/admin_publications?depth=0\u0026page%5Bnumber%5D=2\u0026page%5Bsize%5D=1000\u0026publication_statuses%5B%5D=published\u0026publication_statuses%5B%5D=archived\u0026remove_not_allowed_parents=true',
  prev: null,
  next: 'http://localhost:3000/web_api/v1/admin_publications?depth=0\u0026page%5Bnumber%5D=2\u0026page%5Bsize%5D=1000\u0026publication_statuses%5B%5D=published\u0026publication_statuses%5B%5D=archived\u0026remove_not_allowed_parents=true',
};

export default jest.fn(() => {
  return { data: { pages: [{ data: adminPublicationsData, links }] } };
});
