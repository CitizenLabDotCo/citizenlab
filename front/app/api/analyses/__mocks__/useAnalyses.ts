import { IAnalysisData } from '../types';

export const analysesData: IAnalysisData[] = [
  {
    id: '1',
    type: 'analysis',
    attributes: {
      created_at: '2021-06-01T08:00:00.000Z',
      updated_at: '2021-06-01T08:00:00.000Z',
      participation_method: 'survey',
    },
    relationships: {
      project: {
        data: {
          id: '1',
          type: 'project',
        },
      },
      phase: {
        data: {
          id: '1',
          type: 'phase',
        },
      },
      custom_fields: {
        data: [
          {
            id: '1',
            type: 'custom_field',
          },
        ],
      },
    },
  },
];

export default jest.fn(() => {
  return { data: { data: analysesData } };
});
