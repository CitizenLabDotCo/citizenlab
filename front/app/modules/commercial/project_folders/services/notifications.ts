import { Multiloc } from 'typings';

declare module 'services/notifications' {
  export interface IProjectFolderModerationRightsReceivedNotificationData
    extends IBaseNotificationData {
    attributes: {
      type: 'project_folder_moderation_rights_received';
      read_at: string | null;
      created_at: string;
      project_folder_id: string;
      project_folder_title_multiloc: Multiloc;
    };
  }

  export interface INotificationDataMap {
    IProjectFolderModerationRightsReceivedNotificationData: IProjectFolderModerationRightsReceivedNotificationData;
  }
}
