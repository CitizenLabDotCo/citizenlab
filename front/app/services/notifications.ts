import { API_PATH } from 'containers/App/constants';
import streams, { IStreamParams } from 'utils/streams';

import { authUserStream } from './auth';
import { Multiloc } from 'typings';

const apiEndpoint = `${API_PATH}/notifications`;

export interface IBaseNotificationData {
  id: string;
  type: string;
}

export interface IAdminRightsReceivedNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'admin_rights_received';
    read_at: string | null;
    created_at: string;
  };
}

export interface ICommentDeletedByAdminNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'comment_deleted_by_admin';
    read_at: string | null;
    created_at: string;
    post_type: 'Initiative' | 'Idea';
    post_slug: string | null;
    post_title_multiloc: Multiloc;
    reason_code: 'irrelevant' | 'inappropriate' | 'other';
    other_reason: string;
  };
}

export interface ICommentMarkedAsSpamNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'comment_marked_as_spam';
    read_at: string | null;
    created_at: string;
    initiating_user_first_name: string | null;
    initiating_user_last_name: string | null;
    initiating_user_slug: string | null;
    post_type: 'Initiative' | 'Idea';
    post_slug: string | null;
    post_title_multiloc: Multiloc;
  };
}

export interface ICommentOnYourCommentNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'comment_on_your_comment';
    read_at: string | null;
    created_at: string;
    initiating_user_first_name: string | null;
    initiating_user_last_name: string | null;
    initiating_user_slug: string | null;
    post_type: 'Initiative' | 'Idea';
    post_slug: string | null;
    post_title_multiloc: Multiloc;
  };
}

export interface ICommentOnYourIdeaNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'comment_on_your_idea';
    read_at: string | null;
    created_at: string;
    initiating_user_first_name: string | null;
    initiating_user_last_name: string | null;
    initiating_user_slug: string | null;
    post_slug: string | null;
    post_title_multiloc: Multiloc;
  };
}

export interface ICommentOnYourInitiativeNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'comment_on_your_initiative';
    read_at: string | null;
    created_at: string;
    initiating_user_first_name: string | null;
    initiating_user_last_name: string | null;
    initiating_user_slug: string | null;
    post_slug: string | null;
    post_title_multiloc: Multiloc;
  };
}

export interface IIdeaAssignedToYouNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'idea_assigned_to_you';
    read_at: string | null;
    created_at: string;
    initiating_user_first_name: string | null;
    initiating_user_last_name: string | null;
    initiating_user_slug: string | null;
    post_title_multiloc: Multiloc;
    post_slug: string;
  };
}

export interface IIdeaMarkedAsSpamNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'idea_marked_as_spam';
    read_at: string | null;
    created_at: string;
    initiating_user_first_name: string | null;
    initiating_user_last_name: string | null;
    initiating_user_slug: string | null;
    post_title_multiloc: Multiloc;
    post_slug: string;
  };
}

export interface IInitiativeAssignedToYouNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'initiative_assigned_to_you';
    read_at: string | null;
    created_at: string;
    initiating_user_first_name: string | null;
    initiating_user_last_name: string | null;
    initiating_user_slug: string | null;
    post_title_multiloc: Multiloc;
    post_slug: string;
  };
}

export interface IInitiativeMarkedAsSpamNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'initiative_marked_as_spam';
    read_at: string | null;
    created_at: string;
    initiating_user_first_name: string | null;
    initiating_user_last_name: string | null;
    initiating_user_slug: string | null;
    post_title_multiloc: Multiloc;
    post_slug: string;
  };
}

export interface IInviteAcceptedNotificationData extends IBaseNotificationData {
  attributes: {
    type: 'invite_accepted';
    read_at: string | null;
    created_at: string;
    initiating_user_first_name: string | null;
    initiating_user_last_name: string | null;
    initiating_user_slug: string | null;
  };
}

export interface IMentionInCommentNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'mention_in_comment';
    read_at: string | null;
    created_at: string;
    initiating_user_first_name: string | null;
    initiating_user_last_name: string | null;
    initiating_user_slug: string | null;
    post_type: 'Initiative' | 'Idea';
    post_slug: string | null;
    post_title_multiloc: Multiloc;
  };
}

export interface IMentionInOfficialFeedbackNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'mention_in_official_feedback';
    read_at: string | null;
    created_at: string;
    initiating_user_first_name: string | null;
    initiating_user_last_name: string | null;
    initiating_user_slug: string | null;
    official_feedback_author: Multiloc;
    post_type: 'Initiative' | 'Idea';
    post_slug: string | null;
    post_title_multiloc: Multiloc;
  };
}

export interface IOfficialFeedbackOnCommentedIdeaNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'official_feedback_on_commented_idea';
    read_at: string | null;
    created_at: string;
    official_feedback_author: Multiloc;
    post_slug: string | null;
    post_title_multiloc: Multiloc;
  };
}

export interface IOfficialFeedbackOnCommentedInitiativeNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'official_feedback_on_commented_initiative';
    read_at: string | null;
    created_at: string;
    official_feedback_author: Multiloc;
    post_slug: string | null;
    post_title_multiloc: Multiloc;
  };
}

export interface IOfficialFeedbackOnVotedIdeaNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'official_feedback_on_voted_idea';
    read_at: string | null;
    created_at: string;
    official_feedback_author: Multiloc;
    post_slug: string | null;
    post_title_multiloc: Multiloc;
  };
}

export interface IOfficialFeedbackOnVotedInitiativeNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'official_feedback_on_voted_initiative';
    read_at: string | null;
    created_at: string;
    official_feedback_author: Multiloc;
    post_slug: string | null;
    post_title_multiloc: Multiloc;
  };
}

export interface IOfficialFeedbackOnYourIdeaNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'official_feedback_on_your_idea';
    read_at: string | null;
    created_at: string;
    official_feedback_author: Multiloc;
    post_slug: string | null;
    post_title_multiloc: Multiloc;
  };
}

export interface IOfficialFeedbackOnYourInitiativeNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'official_feedback_on_your_initiative';
    read_at: string | null;
    created_at: string;
    official_feedback_author: Multiloc;
    post_slug: string | null;
    post_title_multiloc: Multiloc;
  };
}

export interface IProjectModerationRightsReceivedNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'project_moderation_rights_received';
    read_at: string | null;
    created_at: string;
    project_id: string;
    project_title_multiloc: Multiloc;
  };
}

export interface IProjectPhaseStartedNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'project_phase_started';
    read_at: string | null;
    created_at: string;
    phase_title_multiloc: Multiloc;
    phase_start_at: string;
    project_slug: string;
    project_title_multiloc: Multiloc;
  };
}

export interface IProjectPhaseUpcomingNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'project_phase_upcoming';
    read_at: string | null;
    created_at: string;
    phase_title_multiloc: Multiloc;
    phase_start_at: string;
    project_slug: string;
    project_title_multiloc: Multiloc;
  };
}

export interface IStatusChangeOfYourIdeaNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'status_change_of_your_idea';
    read_at: string | null;
    created_at: string;
    post_title_multiloc: Multiloc;
    post_slug: string;
    idea_status_title_multiloc: Multiloc;
  };
}

export interface IStatusChangeOfYourInitiativeNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'status_change_of_your_initiative';
    read_at: string | null;
    created_at: string;
    post_title_multiloc: Multiloc;
    post_slug: string;
    initiative_status_title_multiloc: Multiloc;
  };
}

export interface IStatusChangeOnCommentedIdeaNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'status_change_on_commented_idea';
    read_at: string | null;
    created_at: string;
    post_title_multiloc: Multiloc;
    post_slug: string;
    idea_status_title_multiloc: Multiloc;
  };
}

export interface IStatusChangeOnCommentedInitiativeNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'status_change_on_commented_initiative';
    read_at: string | null;
    created_at: string;
    post_title_multiloc: Multiloc;
    post_slug: string;
    initiative_status_title_multiloc: Multiloc;
  };
}

export interface IStatusChangeOnVotedIdeaNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'status_change_on_voted_idea';
    read_at: string | null;
    created_at: string;
    post_title_multiloc: Multiloc;
    post_slug: string;
    idea_status_title_multiloc: Multiloc;
  };
}

export interface IStatusChangeOnVotedInitiativeNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'status_change_on_voted_initiative';
    read_at: string | null;
    created_at: string;
    post_title_multiloc: Multiloc;
    post_slug: string;
    initiative_status_title_multiloc: Multiloc;
  };
}

export interface IThresholdReachedForAdminNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'threshold_reached_for_admin';
    read_at: string | null;
    created_at: string;
    post_title_multiloc: Multiloc;
    post_slug: string;
  };
}

export interface IProjectFolderModerationRightsReceivedNotificationData
  extends IBaseNotificationData {
  attributes: {
    type: 'project_folder_moderation_rights_received';
    read_at: string | null;
    created_at: string;
    project_folder_id: string;
    project_folder_title_multiloc: Multiloc;
  };
}

export interface INotificationDataMap {
  IAdminRightsReceivedNotificationData: IAdminRightsReceivedNotificationData;
  ICommentDeletedByAdminNotificationData: ICommentDeletedByAdminNotificationData;
  ICommentMarkedAsSpamNotificationData: ICommentMarkedAsSpamNotificationData;
  ICommentOnYourCommentNotificationData: ICommentOnYourCommentNotificationData;
  ICommentOnYourIdeaNotificationData: ICommentOnYourIdeaNotificationData;
  ICommentOnYourInitiativeNotificationData: ICommentOnYourInitiativeNotificationData;
  IIdeaAssignedToYouNotificationData: IIdeaAssignedToYouNotificationData;
  IIdeaMarkedAsSpamNotificationData: IIdeaMarkedAsSpamNotificationData;
  IInitiativeAssignedToYouNotificationData: IInitiativeAssignedToYouNotificationData;
  IInitiativeMarkedAsSpamNotificationData: IInitiativeMarkedAsSpamNotificationData;
  IInviteAcceptedNotificationData: IInviteAcceptedNotificationData;
  IMentionInCommentNotificationData: IMentionInCommentNotificationData;
  IMentionInOfficialFeedbackNotificationData: IMentionInOfficialFeedbackNotificationData;
  IOfficialFeedbackOnCommentedIdeaNotificationData: IOfficialFeedbackOnCommentedIdeaNotificationData;
  IOfficialFeedbackOnCommentedInitiativeNotificationData: IOfficialFeedbackOnCommentedInitiativeNotificationData;
  IOfficialFeedbackOnVotedIdeaNotificationData: IOfficialFeedbackOnVotedIdeaNotificationData;
  IOfficialFeedbackOnVotedInitiativeNotificationData: IOfficialFeedbackOnVotedInitiativeNotificationData;
  IOfficialFeedbackOnYourIdeaNotificationData: IOfficialFeedbackOnYourIdeaNotificationData;
  IOfficialFeedbackOnYourInitiativeNotificationData: IOfficialFeedbackOnYourInitiativeNotificationData;
  IProjectModerationRightsReceivedNotificationData: IProjectModerationRightsReceivedNotificationData;
  IProjectPhaseStartedNotificationData: IProjectPhaseStartedNotificationData;
  IProjectPhaseUpcomingNotificationData: IProjectPhaseUpcomingNotificationData;
  IStatusChangeOfYourIdeaNotificationData: IStatusChangeOfYourIdeaNotificationData;
  IStatusChangeOfYourInitiativeNotificationData: IStatusChangeOfYourInitiativeNotificationData;
  IStatusChangeOnCommentedIdeaNotificationData: IStatusChangeOnCommentedIdeaNotificationData;
  IStatusChangeOnCommentedInitiativeNotificationData: IStatusChangeOnCommentedInitiativeNotificationData;
  IStatusChangeOnVotedIdeaNotificationData: IStatusChangeOnVotedIdeaNotificationData;
  IStatusChangeOnVotedInitiativeNotificationData: IStatusChangeOnVotedInitiativeNotificationData;
  IThresholdReachedForAdminNotificationData: IThresholdReachedForAdminNotificationData;
  IProjectFolderModerationRightsReceivedNotificationData: IProjectFolderModerationRightsReceivedNotificationData;
}

export type TNotificationData =
  INotificationDataMap[keyof INotificationDataMap];

export interface INotificationLinks {
  self: string;
  first: string;
  prev: string;
  next: string;
  last: string;
}

export interface INotifications {
  data: TNotificationData[];
  links: INotificationLinks;
}

export function notificationsStream(streamParams: IStreamParams | null = null) {
  return streams.get<INotifications>({ apiEndpoint, ...streamParams });
}

export async function markAllAsRead() {
  const response = await streams.add(`${apiEndpoint}/mark_all_read`, null);
  await authUserStream().fetch();
  return response;
}
