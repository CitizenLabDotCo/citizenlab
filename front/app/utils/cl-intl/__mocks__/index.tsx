import React from 'react';
const Intl = jest.requireActual('react-intl');

// Initialise the real provider so that we don't
// need to reimplement any internals
const defaultProps = {
  locale: 'en',
  defaultLocale: 'en',
};
const intlProvider = new Intl.IntlProvider(defaultProps, {});

// The exact same `intl` object the real code receives ;-)
const { intl } = intlProvider.getChildContext();

const formatMessageReplacement = (
  messageDescriptor: ReactIntl.FormattedMessage.MessageDescriptor,
  values?: { [key: string]: string | number | boolean | Date } | undefined
) => {
  return intl.formatMessage(messageDescriptor, {
    tenantName: 'The Test',
    orgName: 'Test Town',
    orgType: 'testing',
    ...(values || {}),
  });
};

const intlReplacement = {
  ...intl,
  formatMessage: formatMessageReplacement,
};

const injectIntlReplacement = (Node) => {
  const renderWrapped = (props) => <Node {...props} intl={intl} />;
  renderWrapped.displayName = Node.displayName || Node.name || 'Component';
  return renderWrapped;
};

Intl.injectIntl = injectIntlReplacement;

// Override components by implementing the real components
//  providing them the context they need in order to function
const {
  IntlProvider,
  FormattedDate,
  FormattedTime,
  FormattedRelative,
  FormattedNumber,
  FormattedPlural,
  FormattedMessage,
  FormattedHTMLMessage,
} = Intl;

Intl.FormattedDate = (props) => (
  <IntlProvider {...defaultProps}>
    <FormattedDate {...props} />
  </IntlProvider>
);
Intl.FormattedTime = (props) => (
  <IntlProvider {...defaultProps}>
    <FormattedTime {...props} />
  </IntlProvider>
);
Intl.FormattedRelative = (props) => (
  <IntlProvider {...defaultProps}>
    <FormattedRelative {...props} />
  </IntlProvider>
);
Intl.FormattedNumber = (props) => (
  <IntlProvider {...defaultProps}>
    <FormattedNumber {...props} />
  </IntlProvider>
);
Intl.FormattedPlural = (props) => (
  <IntlProvider {...defaultProps}>
    <FormattedPlural {...props} />
  </IntlProvider>
);
Intl.FormattedMessage = (props) => (
  <IntlProvider {...defaultProps}>
    <FormattedMessage {...props} />
  </IntlProvider>
);
Intl.FormattedHTMLMessage = (props) => (
  <IntlProvider {...defaultProps}>
    <FormattedHTMLMessage {...props} />
  </IntlProvider>
);

// Set displayName so that snapshots don't use "Uknown" as component name
Intl.FormattedDate.displayName = 'FormattedDate';
Intl.FormattedTime.displayName = 'FormattedTime';
Intl.FormattedRelative.displayName = 'FormattedRelative';
Intl.FormattedNumber.displayName = 'FormattedNumber';
Intl.FormattedPlural.displayName = 'FormattedPlural';
Intl.FormattedMessage.displayName = 'FormattedMessage';
Intl.FormattedHTMLMessage.displayName = 'FormattedHTMLMessage';

// Special hook for tests, real package does not export this
Intl.intl = intlReplacement;

module.exports = Intl;
