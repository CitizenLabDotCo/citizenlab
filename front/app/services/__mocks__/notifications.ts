import { INotifications, TNotificationData } from 'services/notifications';
import { BehaviorSubject } from 'rxjs';

let mockNotifications: INotifications[] | null = null;

export const __setMockNotifications = (notifications: INotifications[]) => {
  mockNotifications = notifications;
};

export const notificationsStream = jest.fn((_notification) => {
  const observable = new BehaviorSubject(mockNotifications);
  return {
    observable,
  };
});

export const getMockNotifications = (notifs: TNotificationData[], isLast) => ({
  data: notifs,
  links: {
    fist: 'first',
    self: 'self',
    prev: 'prev',
    last: 'last',
    next: isLast ? null : 'next',
  },
});

export const getNotification = (
  id: string,
  type:
    | 'project_phase_upcoming'
    | 'project_phase_started'
    | 'mention_in_official_feedback'
    | 'official_feedback_on_commented_idea'
    | 'official_feedback_on_voted_idea'
    | 'official_feedback_on_your_idea'
    | 'new_idea_for_admin'
    | 'admin_rights_received'
    | 'project_moderation_rights_received'
    | 'new_comment_for_admin'
    | 'comment_on_your_comment'
    | 'comment_on_your_idea'
    | 'mention_in_comment'
    | 'idea_marked_as_spam'
    | 'comment_marked_as_spam'
    | 'status_change_of_your_idea'
    | 'invite_accepted'
    | 'comment_deleted_by_admin',
  readAt?: undefined | string
) =>
  ({
    id,
    type: 'notification',
    attributes: {
      type,
      read_at: readAt ? readAt : null,
      created_at: '2019-04-16T14:32:17.095Z',
      initiating_user_first_name:
        ['status_change_of_your_idea', 'admin_rights_received'].indexOf(
          type
        ) === -1
          ? 'Initiating'
          : undefined,
      initiating_user_last_name:
        ['status_change_of_your_idea', 'admin_rights_received'].indexOf(
          type
        ) === -1
          ? 'User'
          : undefined,
      initiating_user_slug:
        ['status_change_of_your_idea', 'admin_rights_received'].indexOf(
          type
        ) === -1
          ? 'initiating-user'
          : undefined,
      idea_title: {
        'nl-BE': 'Qui nihil veniam consequatur.',
      },
      reason_code:
        ['comment_deleted_by_admin'].indexOf(type) === -1
          ? undefined
          : 'irrelevant',
    },
    relationships: {
      // recipient: {
      //   data: {
      //     id: '9aae2c12-859d-4abb-ac80-80c005459343',
      //     type: 'users'
      //   }
      // },
      initiating_user: {
        data: {
          id: 'fe947bd3-9c67-40ed-96a3-9a12c4c73b46',
          type: 'user',
        },
      },
      post:
        ['invite_accepted', 'admin_rights_received'].indexOf(type) === -1
          ? {
              data: {
                id: '5dda9f3e-4c3f-45f1-bb09-b88a40776cab',
                type: 'idea',
              },
            }
          : undefined,
      comment:
        ['admin_rights_received', 'invite_accepted'].indexOf(type) === -1
          ? {
              data: {
                id: '13d121b7-ee79-4a17-ac3e-c4ee51beb980',
                type: 'comment',
              },
            }
          : undefined,
      idea_status:
        ['status_change_of_your_idea'].indexOf(type) !== -1
          ? {
              data: {
                id: 'status',
                type: 'idea_status',
              },
            }
          : undefined,
      invite:
        ['invite_accepted'].indexOf(type) !== -1
          ? {
              data: {
                id: 'iviteId',
                type: 'invite',
              },
            }
          : undefined,
      project:
        [
          'comment_on_your_idea',
          'invite_accepted',
          'admin_rights_received',
        ].indexOf(type) === -1
          ? {
              data: {
                id: '747dcdb1-f547-409f-8b90-8296d36d560a',
                type: 'project',
              },
            }
          : undefined,
    },
  } as TNotificationData);
