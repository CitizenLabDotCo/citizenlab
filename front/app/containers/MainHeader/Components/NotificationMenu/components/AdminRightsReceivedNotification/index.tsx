import React from 'react';

import { IAdminRightsReceivedNotificationData } from 'api/notifications/types';

import { FormattedMessage } from 'utils/cl-intl';

import messages from '../../messages';
import NotificationWrapper from '../NotificationWrapper';

interface Props {
  notification: IAdminRightsReceivedNotificationData;
}

const AdminRightsReceivedNotification = ({ notification }: Props) => {
  return (
    <NotificationWrapper
      linkTo={'/admin'}
      timing={notification.attributes.created_at}
      icon="shield-checkered"
      isRead={!!notification.attributes.read_at}
    >
      <FormattedMessage {...messages.adminRightsReceived} />
    </NotificationWrapper>
  );
};

export default AdminRightsReceivedNotification;
