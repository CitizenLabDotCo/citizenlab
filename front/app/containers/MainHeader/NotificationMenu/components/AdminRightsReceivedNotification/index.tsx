import React, { memo } from 'react';
import { IAdminRightsReceivedNotificationData } from 'services/notifications';
import messages from '../../messages';
import { FormattedMessage } from 'utils/cl-intl';
import NotificationWrapper from '../NotificationWrapper';

interface Props {
  notification: IAdminRightsReceivedNotificationData;
}

const AdminRightsReceivedNotification = memo<Props>((props) => {
  const { notification } = props;

  return (
    <NotificationWrapper
      linkTo={'/admin'}
      timing={notification.attributes.created_at}
      icon="admin"
      isRead={!!notification.attributes.read_at}
    >
      <FormattedMessage {...messages.adminRightsReceived} />
    </NotificationWrapper>
  );
});

export default AdminRightsReceivedNotification;
