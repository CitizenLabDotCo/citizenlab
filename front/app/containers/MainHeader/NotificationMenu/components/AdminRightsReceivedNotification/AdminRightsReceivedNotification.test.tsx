// @ts-nocheck
// libraries
import React from 'react';
import { shallow } from 'enzyme';

// component to test
import AdminRightsReceivedNotification from './';

// mock utilities
jest.mock('utils/cl-intl');
jest.mock('utils/analytics', () => ({ trackEventByName: () => {} }));
jest.mock('modules', () => ({ streamsToReset: [] }));

import { getNotification } from 'services/__mocks__/notifications';

describe('<AdminRightsReceivedNotification />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <AdminRightsReceivedNotification
        notification={getNotification('aminRights', 'admin_rights_received')}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
