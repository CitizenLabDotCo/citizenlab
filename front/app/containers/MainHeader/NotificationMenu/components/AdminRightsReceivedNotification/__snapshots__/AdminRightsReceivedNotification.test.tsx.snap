// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminRightsReceivedNotification /> renders correctly 1`] = `
<NotificationWrapper
  icon="admin"
  isRead={false}
  linkTo="/admin"
  timing="2019-04-16T14:32:17.095Z"
>
  <FormattedMessage
    defaultMessage="You're now an administrator of the platform"
    id="app.containers.NotificationMenu.adminRightsReceived"
  />
</NotificationWrapper>
`;
